specs/agents/tracing-instrumentation-db.md

```diff
@@ -1,34 +1,198 @@
+| <hr/> |<hr/>|<hr/>|
+|`context.db.instance`:red_circle:| Database instance name, e.g. "customers". For DynamoDB, this is the region.| :x: |
+|`context.db.statement`| Statement/query, e.g. `SELECT * FROM foo WHERE ...`. The full database statement should be stored in db.statement, which may be useful for debugging performance issues. We store up to 10000 Unicode characters per database statement. For Non-SQL data stores see details below.| :x: |
+|`context.db.type`| Database type/category, which should be "sql" for SQL databases, and the lower-cased database name otherwise.| :x: |
+|`context.db.user`| Username used for database access, e.g. `readonly_user`| :x: |
+|`context.db.link`:red_circle:| ??? | :x: |
+|`context.db.rows_affected`| The number of rows / entities affected by the corresponding db statement / query.| :x: |
+| <hr/> |<hr/>|<hr/>|
+|`context.destination.address`|The hostname / address of the database.| :x: |
+|`context.destination.port`|The port under which the database is accessible.| :x: |
+|`context.destination.service.name`| The `destination.service.name` is used to denote "sameness" of the service. E.g. multiple instances of Oracle databases have all the same name `oracle`. For databases and storages the same value as for the `span.subtype` should be used.| :white_check_mark:|
+|`context.destination.service.type`| Should be the same as the `span.type`. Value: `db`| :white_check_mark:|
+|`context.destination.service.resource`:red_circle:|  Used to detect unique destinations from each service. This field should contain all information that is needed to differentiate different database / storage instances (e.g. in the service map). See details below on how to set this field for specific technologies.| :white_check_mark:|
+|`context.destination.cloud.region`| The cloud region in case the datastore is hosted in a public cloud or is a managed datasatore / database. E.g. AWS regions, such as `us-east-1` | :x: |
 
-For SQL databases this will be the full SQL statement.
 
-For MongoDB, this can be set to the command encoded as MongoDB Extended JSON.
+## Specific Databases
 
-For Elasticsearch search-type queries, the request body may be recorded. Alternatively, if a query is specified in HTTP query parameters, that may be used instead. If the body is gzip-encoded, the body should be decoded first.
+### AWS DynamoDb
 
-### Database span names
+<details>
+  <summary>Click to expand</summary>
 
-For SQL operations we perform a limited parsing the statement, and extract the operation name and outer-most table involved (if any). See more details here: https://docs.google.com/document/d/1sblkAP1NHqk4MtloUta7tXjDuI_l64sT2ZQ_UFHuytA.
+| Field | Value / Examples | Comments |
+|-------|:---------------:|----------|
+|`name`| e.g. `DynamoDB UpdateItem my_table`|  The span name should capture the operation name in CamelCase and the table name, if available. The format should be `DynamoDB <ActionName> <TableName>`. |
```

Comment:
> CamelCase

It is less about being CamelCase than it is about using the same naming for those actions as AWS.